apps/angular/projection/src/app/component/teacher-card/teacher-card.component.ts

```diff
@@ -1,31 +1,59 @@
-import { Component, OnInit } from '@angular/core';
-import { FakeHttpService } from '../../data-access/fake-http.service';
+import { AsyncPipe } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnInit,
+  ViewEncapsulation,
+} from '@angular/core';
+import { Observable } from 'rxjs';
+import {
+  FakeHttpService,
+  randTeacher,
+} from '../../data-access/fake-http.service';
 import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
 import { Teacher } from '../../model/teacher.model';
 import { CardComponent } from '../../ui/card/card.component';
 
 @Component({
   selector: 'app-teacher-card',
   template: `
     <app-card
-      [list]="teachers"
-      [type]="cardType"
-      customClass="bg-light-red"></app-card>
+      [list]="teachers$ | async"
+      [content]="'firstName'"
+      [imgTemplate]="imgTemplate"
+      [buttonTemplate]="buttonTemplate"
+      [deleteTemplate]="deleteTemplate"
+      class="bg-light-red"></app-card>
+    <ng-template #imgTemplate>
+      <img src="assets/img/teacher.png" width="200px" />
+    </ng-template>
+    <ng-template #buttonTemplate>
+      <button
+        class="rounded-sm border border-blue-500 bg-blue-300 p-2"
+        (click)="addNewItem()">
+        Add
+      </button>
+    </ng-template>
+    <ng-template #deleteTemplate let-id>
+      <button (click)="delete(id)">
+        <img class="h-5" src="assets/svg/trash.svg" />
+      </button>
+    </ng-template>
   `,
   styles: [
     `
-      ::ng-deep .bg-light-red {
+      .bg-light-red {
         background-color: rgba(250, 0, 0, 0.1);
       }
     `,
   ],
   standalone: true,
-  imports: [CardComponent],
+  imports: [CardComponent, AsyncPipe],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class TeacherCardComponent implements OnInit {
-  teachers: Teacher[] = [];
-  cardType = CardType.TEACHER;
+  teachers$: Observable<Teacher[]> = this.store.teachers$;
```

Comment:
You should use signal here. Get used to the new angular way of binding properties to the tempalte